lib/chef/resource/inspec_waiver_file.rb

```diff
@@ -0,0 +1,152 @@
+          expiration '2021-01-01'
+          action :add
+        end
+      ```
+
+      **Add an InSpec waiver entry to a given waiver file using the 'name' property to identify the control**:
+
+      ```ruby
+        inspec_waiver_file 'my_inspec_control_01' do
+          file 'C:\\chef\\inspec_waiver.yml'
+          justification "The subject of this control is not managed by Chef on the systems in policy group \#{node['policy_group']}"
+          action :add
+        end
+      ```
+
+      **Remove an InSpec waiver entry to a given waiver file**:
+
+      ```ruby
+        inspec_waiver_file "my_inspec_control_01" do
+          file '/etc/chef/inspec_waiver.yml'
+          action :remove
+        end
+      ```
+      DOC
+
+      property :control, String,
+        name_property: true,
+        description: "The name of the control being added or removed to the waiver file"
+
+      property :file, String,
+        required: true,
+        description: "The path to the waiver file being modified"
+
+      property :expiration, String,
+        description: "The expiration date of the given waiver - provided in YYYY-MM-DD format",
+        callbacks: {
+          "Expiration date should match the following format: YYYY-MM-DD" => proc { |e|
+            re = Regexp.new('([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))').freeze
+            if re.match?(e) || e.nil?
+              true
+            else
+              false
+            end
+          },
+          }
+
+      property :run_test, [true, false],
+        description: "If present and true, the control will run and be reported, but failures in it won’t make the overall run fail. If absent or false, the control will not be run."
+
+      property :justification, String,
+        description: "Can be any text you want and might include a reason for the waiver as well as who signed off on the waiver."
+
+      property :backup, [false, Integers],
+        description: "The number of backups to be kept in /var/chef/backup (for UNIX- and Linux-based platforms) or C:/chef/backup (for the Microsoft Windows platform). Set to false to prevent backups from being kept.",
+        default: false
+
+      action :add do
+        filename = new_resource.file
+        yaml_contents = if ::File.file?(filename) && ::File.readable?(filename) && !::File.zero?(filename)
+                          IO.read(new_resource.file)
+                        else
+                          ""
+                        end
+        waiver_hash = {}
+        waiver_hash = ::YAML.safe_load(yaml_contents) unless yaml_contents.empty?
+        control_hash = {}
+        control_hash["expiration_date"] = new_resource.expiration.to_s unless new_resource.expiration.nil?
+        control_hash["run"] = new_resource.run_test unless new_resource.run_test.nil?
+        control_hash["justification"] = new_resource.justification.to_s
+
+        if waiver_hash.key?("#{new_resource.control}")
```

Comment:
```suggestion
        if waiver_hash.key?(new_resource.control)
```

Since this is a string, the interpolation is unnecessary.